client/me/purchases/utils.js

```diff
@@ -62,12 +63,36 @@ function recordPageView( trackingSlug, props, nextProps = null ) {
 	analytics.tracks.recordEvent( `calypso_${ trackingSlug }_purchase_view`, { product_slug: productSlug } );
 }
 
+function enrichedSurveyData( surveyData, moment, site, purchase ) {
+	const purchaseStartDate = get( purchase, 'subscribedDate', null );
+	const siteStartDate = get( site, 'options.created_at', null );
+	const purchaseId = get( purchase, 'id', null );
+	const productSlug = get( purchase, 'productSlug', null );
+
+	const enrichment = {
+		purchase: productSlug,
+		purchaseId
+	};
+
+	if ( purchaseStartDate ) {
+		enrichment.daysSincePurchase = moment.diff( purchaseStartDate, 'days', true );
+		enrichment.newOrRenewal = moment.diff( purchaseStartDate, 'years', true ) > 1.0 ? 'renewal' : 'new purchase';
+	}
+
+	if ( siteStartDate ) {
+		enrichment.daysSinceSiteCreation = moment.diff( siteStartDate, 'days', true );
+	}
+
+	return { ...surveyData, ...enrichment };
+}
```

Comment:
fun fact! we can also combine this into one big declarative expression with `Object.assign()`

```js
return Object.assign(
	{
		purchase: productSlug,
		purchaseId,
	},
	purchaseStartDate && {
		daysSincePurchase: moment.diff( … ),
		newOrRenewal: …,
	},
	siteStartDate && {
		daysSinceSiteCreation: …
	}
);
```